### DISCUSSION TITLE
Is there a way to exclude some text from getting "tooltipped" by my glossary?

### DISCUSSION THREAD

**Discussion by feasgal** [squidfunk/mkdocs-material#5139] [Q&A] [answered]
We are using snippets to auto-append a glossary file. The tooltips are case-sensitive, so we capitalize the words in the glossary file, and then if we want the word to have a tooltip, we capitalize it.

So for instance in our `glossary.md`, I have:

```
*[Product]: A set of information that...

*[Group]: Products can be put into groups for organization and...
```

And then in the docs, if we want to define a word, we would capitalize it:

```
To make a Group, you will need at least one Product defined in...
```

This occasionally causes us to get a tooltip where we didn't really need one, like at the beginning of a sentence, but that's not a big deal.

But in a totally different section of the docs, I have to document a part of a UI that has a table entitled 'Product Group,' but has nothing to do with those definitions of 'Product' and 'Group'. I need to talk about "the Product Group table," and I can't just not capitalize it because it's the title of the table, but obviously it's getting those two tooltips that are not just unwanted, but inaccurate in this context.

Anyone know a clever way to avoid those individual words getting wrapped in `<abbr>` elements?

**Comment by LandonTClipp** [comment]
I think the glossary was originally intended to be for abbreviations and acronyms (i.e. things that wouldn't normally be confused with other words).

**Reply by feasgal** [reply]
Oh, for sure. The way we're using it is not quite as intended, hence the problem we're running into. That's why I put this here in Discussions, not over in Issues. Not a request, so much as a "has someone maybe thought of something clever that I haven't?"

**Comment by squidfunk** [comment, marked as answer]
There's currently no way to do so, no. The limitations lie not within Material for MkDocs, but within the [Abbreviations](https://python-markdown.github.io/extensions/abbreviations/) Markdown extension which will automatically decorate matching words. The change would somehow need to be made in the way the Markdown parser identifies those words, so I recommend taking this upstream to [Python Markdown](https://github.com/Python-Markdown/markdown/).

We're evaluating adding better glossary support to Material for MkDocs in the coming months. The idea is to use a combination of instant previews (incubating) and auto-linking, but we first need to refactor instant loading addressing #3797 and #5023. I'm on it.

**Comment by karenmajewicz** [comment]
Here is a related question: Is there a way to disable the Abbreviations for a selected page?

My team would like to have both the tooltip style glossary, as currently documented in the [Tooltips section.](https://squidfunk.github.io/mkdocs-material/reference/tooltips/#adding-a-glossary), as well as a dedicated page listing terms and definitions. 

To do this, I created

1.  `abbreviations.md` file using the format `*[term]: definitition`. (This format does not render as a regular HTML page)
2. a regular markdown page (`glossary.md`) with the same list, but formatted with headers and text.

With this method, the tooltips appear on every entry in `glossary.md.` This duplicates the definitions and makes the page seem cluttered.  Is there is a way to suppress the auto_append tooltips for just one page? Or, is there another workaround here that I have not considered?

**Reply by kamilkrzyskow** [reply]
I believe there is no built-in toggle that would allow to disable a markdown extension on a given page, and Snippets don't have an exclude feature. However you could write a simple [hook](https://www.mkdocs.org/user-guide/configuration/#hooks), which for a given page, before it's rendered, modifies the global markdown_extension config (set in `mkdocs.yml`) and then after it's rendered restores the config to how it was initially. You'd use the `on_page_markdown` and `on_page_content` [events](https://www.mkdocs.org/dev-guide/plugins/#events). 
Let me know if you'd need more help.
> Reference for modifying the markdown extensions ~~config~~: https://github.com/mkdocs/mkdocs/discussions/3196#discussioncomment-5703426 _edit: actually it doesn't show how to edit the config specifically_ 🤦
> _edit2_: The tags plugin has an example how to access the [config dict](https://github.com/squidfunk/mkdocs-material/blob/39507ace3e41f475aa0a7bb859349c63c28ab74d/src/plugins/tags/plugin.py#L60-L61)

**Reply by squidfunk** [reply]
This is something to be taken to the authors of Snippets, not Material for MkDocs.

However, with the new [instant previews](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#instant-previews) feature, we're investigating to build glossary functionality as part of a plugin, which will allow for much more fine-grained control. Feel free to test instant previews and share your requirements for building glossaries in #6704.

**Reply by akhilub** [reply]
@karenmajewicz Yes there is a way to disable the Abbreviations for selected pages, it's a quite lengthy process I implemented it today.
You can check out it here my [glossary](https://dsabible.com/glossary/definitions/#arrays-strings) where Abbreviations are disabled and they (substring, palindromic substring) are enabled on other pages.

Search for them using the search box.
